### PR TITLE
Add posix_utils.getDiskUsage template

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Add `posix_utils.getDiskUsage` convenience proc for `statvfs` to get disk usage statistics in bytes `(total: 23763712, used: 957632, free: 666)`.

:slightly_smiling_face:

### CODE REVIEW

File: lib/posix/posix_utils.nim [hunk 1]
```diff
@@ -99,3 +99,14 @@ proc mkdtemp*(prefix: string): string =
     raise newException(OSError, $strerror(errno))
   return $tmpl
 
+when defined(amd64):
```
Comment (Araq) [inline lib/posix/posix_utils.nim]: So this implementation is only for amd64? Come on, we need to support it on every Posix system.

Comment (juancarlospaco) [reply lib/posix/posix_utils.nim]: 32 Bit does work Ok, but may overflow tiny 32 Bit `uint` really easy being bytes, even if divided,
old test pass confirm that failing on 32 Bit, so I intentionally made it 64 Bit only, test pass 64 Bit.
:slightly_smiling_face:

**Comment by juancarlospaco** [comment]
I agree, in fact it is not on `posix`, it is on `posix_utils` which are by definition *utils*,
module literally says *A set of helpers around Posix*.
:slightly_smiling_face:

File: lib/posix/posix_utils.nim (hunk 1 above)
Comment (Araq) [reply lib/posix/posix_utils.nim]: 1. The check could look like ``when sizeof(int) == 8`` as there are other 64 bit target CPUs. (But it's the wrong idea regardless.)
2. Please use the ``uint64`` type.

Comment (juancarlospaco) [reply lib/posix/posix_utils.nim]: @Araq Done. **Thanks for the help**, sounds totally right. :+1:

File: lib/posix/posix_utils.nim [hunk 2]
```diff
@@ -99,3 +99,14 @@ proc mkdtemp*(prefix: string): string =
...
+
+proc getDiskUsage*(path: string): tuple[total: uint64, used: uint64, free: uint64] {.inline.} =
+  ## Convenience proc for ``statvfs`` to get disk usage statistics in bytes.
+  ## This proc is available for 64 Bit architectures and operating systems.
+  ##
+  ## .. code-block:: nim
+  ##   echo getDiskUsage(".") ## (total: ..., used: ..., free: ...)
+  ##
+  var t = Statvfs()
+  discard statvfs(path, t)
```
Comment (dom96) [inline lib/posix/posix_utils.nim]: This returns an error I think, right? Would be better to check errno and throw it if the function call failed.

Comment (juancarlospaco) [reply lib/posix/posix_utils.nim]: @dom96 Fixed. Thanks for the Help!. I am more used to `doAssert` so I hope the `if` is Ok.

File: lib/posix/posix_utils.nim [hunk 3]
```diff
@@ -99,3 +99,14 @@ proc mkdtemp*(prefix: string): string =
     raise newException(OSError, $strerror(errno))
   return $tmpl
 
+
+proc getDiskUsage*(path: string): tuple[total: uint64, used: uint64, free: uint64] {.inline.} =
+  ## Convenience proc for ``statvfs`` to get disk usage statistics in bytes.
+  ## This proc is available for 64 Bit architectures and operating systems.
```
Comment (Clyybber) [inline lib/posix/posix_utils.nim]: Outdated comment

File: tests/stdlib/tposix.nim [hunk 4]
```diff
@@ -17,3 +17,16 @@ when not defined(windows):
   writeLine(stdout, u.nodename)
   writeLine(stdout, u.release)
   writeLine(stdout, u.machine)
+
+
+  when defined(amd64):
```
Comment (Clyybber) [inline tests/stdlib/tposix.nim]: Doesn't have to be restricted to 64bit anymore

File: lib/posix/posix_utils.nim [hunk 5]
```diff
@@ -99,3 +99,14 @@ proc mkdtemp*(prefix: string): string =
...
+  ##
+  ## .. code-block:: nim
+  ##   echo getDiskUsage(".") ## (total: ..., used: ..., free: ...)
+  ##
+  var t = Statvfs()
+  if statvfs(path, t) != 0.cint: raise newException(OSError, $strerror(errno))
+  (total: uint64(t.f_blocks * t.f_frsize), used: uint64((t.f_blocks - t.f_bfree) * t.f_frsize), free: uint64(t.f_bavail * t.f_frsize))
```
Comment (Clyybber) [inline lib/posix/posix_utils.nim]: `uint64(x * y)` should be `x.uint64 * y.uint64` so as to prevent against overflows during multiplication on 32bit.

Comment (juancarlospaco) [reply lib/posix/posix_utils.nim]: Totally correct.

**Comment by juancarlospaco** [comment]
@Araq Close or Merge as you see appropriate, no problem, is ready.

**Comment by Araq** [comment]
Doesn't look future-proof to me. This begs to use a bignum library IMHO.

**Pull request closed without merging**